src/pybind/mgr/dashboard/frontend/e2e/cluster/hosts.po.ts

```diff
@@ -1,5 +1,49 @@
+import { by, element } from 'protractor';
 import { PageHelper } from '../page-helper.po';
 
 export class HostsPageHelper extends PageHelper {
   pages = { index: '/#/hosts' };
+
+  // function that checks all services links work for first
+  // host in table (if a host is present)
+  check_services_links() {
+    this.navigateTo();
+    let links_tested = 0;
+    this.getTableCount()
+      .getText()
+      .then((hostcount) => {
+        // Only check for services links if there's at least 1 host present
+        if (hostcount.includes('0 total')) {
```

Comment:
We should always test this. If there are no hosts then we have a problem somewhere.
```suggestion
```